mlrun/datastore/targets.py

```diff
@@ -1259,6 +1262,17 @@ def as_df(
     ):
         raise NotImplementedError()
 
+    @property
+    def source_attributes(self) -> dict:
+        keys = ["url", "user", "database", "db_schema", "warehouse"]
+        if self.attributes:
+            snowflake_dict = {key: self.attributes.get(key) for key in keys}
+        else:
+            snowflake_dict = {key: None for key in keys}
```

Comment:
Redundant safeguard:
```suggestion
        snowflake_dict = {key: self.attributes.get(key) for key in keys}
```

Reply:
self.attributes can be None if it is built from dict... lets use the condition

Reply:
It can't.

Reply:
Ok, so
```suggestion
attributes = self.attributes or {}
snowflake_dict = {key: attributes.get(key) for key in keys}
```